src/app.tsx

```diff
@@ -1,44 +1,13 @@
-import React, { useContext } from 'react'
-import Config from './components/config.jsx'
-import { ConfigContext } from './context/config-context.jsx'
-import HelperUi from './helper-ui.jsx'
-import { isConfigPage } from './lib/is-config-page.js'
-import { isPathOrSubdomainRequest, isSubdomainGatewayRequest } from './lib/path-or-subdomain.js'
-import RedirectPage from './redirectPage.jsx'
+import React, { Suspense } from 'react'
+import { RouteContext } from './context/router-context.jsx'
+import './app.css'
 
 function App (): JSX.Element {
-  const { isConfigExpanded, setConfigExpanded } = useContext(ConfigContext)
-  const isRequestToViewConfigPage = isConfigPage(window.location.hash)
-  const isSubdomainRender = isSubdomainGatewayRequest(window.location)
-  const shouldRequestBeHandledByServiceWorker = isPathOrSubdomainRequest(window.location) && !isRequestToViewConfigPage
-
-  if (shouldRequestBeHandledByServiceWorker) {
-    if (window.self === window.top && isSubdomainRender) {
-      return (<RedirectPage />)
-    } else {
-      // rendering redirect page without iframe because this is a top level window and subdomain request.
-      return (<RedirectPage showConfigIframe={false} />)
-    }
-  }
-
-  if (isRequestToViewConfigPage) {
-    if (isSubdomainRender && window.self === window.top) {
-      return <RedirectPage />
-    }
-
-    setConfigExpanded(true)
-    return <Config />
-  }
-
-  if (isSubdomainRender) {
-    return (<RedirectPage />)
-  }
-
-  if (isConfigExpanded) {
-    return (<Config />)
-  }
+  const { currentRoute } = React.useContext(RouteContext)
   return (
-    <HelperUi />
+    <Suspense fallback={<div>Loading...</div>}>
+      {currentRoute?.component != null && <currentRoute.component />}
+    </Suspense>
```

Comment:
we could avoid any potential "not prod ready" issues with preact's `Suspense` by doing a simple useEffect wrapper of the async operation